packages/ketcher-react/src/script/ui/views/components/ContextMenu/ContextMenuTrigger.tsx

```diff
@@ -45,9 +53,18 @@ const ContextMenuTrigger: React.FC<PropsWithChildren> = ({ children }) => {
           functionalGroup.relatedSGroupId,
           functionalGroup
         )
+
+      const sGroupId = struct.sgroups.find((_, sGroup) =>
+        sGroup.atoms.includes(atomId)
+      )
+
+      sGroupId !== null && selectedSGroupsIds.add(sGroupId)
     })
 
-    return selectedFunctionalGroups
+    return {
```

Comment:
if the method now returns functional and sgroup, maybe rename the function? For example, just "getSelectedGroups"

Reply:
Renamed to `getSelectedGroupsInfo`